src/components/structures/MatrixChat.js

```diff
@@ -304,6 +305,9 @@ module.exports = React.createClass({
                 });
                 break;
             case 'view_room':
+                // by default we autoPeek rooms, unless we were called explicitly with
+                // autoPeek=false by something like RoomDirectory who has already peeked
+                this.setState({ autoPeek : payload.auto_peek === false ? false : true });
```

Comment:
This conditional makes no sense. I assume you're trying to catch and cast undefined / null. Just do `Boolean(payload.auto_peek)` then. As it stands, you're NOPing because of the triple equals.